### ISSUE TITLE
problem with ADS after update to 14.3

### ISSUE THREAD

**Issue by karen20ced4** [WWBN/AVideo#9246]
after update from version 12.4 to the last version 14.3, ( i also update DB and plugins from plugin menu)
No showing ads(simple images with link, or html code), especially with sideRectangle position , on the video page 

i check the source code of the page and where it showld be the ad (in my test: Aloha side rectangle  ) i have this message:

**getAdsSideRectangle is empty videos_id=39, not from live videos_id=39, type=sideRectangle, label=sideRectangle [1] [sideRectangle], adCode is empty </div>**

can you help me to understand where can be the problem ? i already read this : https://github.com/WWBN/AVideo/wiki/ADs-plugin, no help. 

i have another AD (simple image with link) on position leaderBoardBigVideo, and that is working fine.
i don't understand. thanks for help 

![sideRectangle](https://github.com/user-attachments/assets/5d3b2790-3a50-4c73-b735-2c7c95eab5dd)

**Comment by DanielnetoDotCom** [comment]
Check your avideo.log, and also inspect the html code where the ads should be, you may see a comment or some HTML to give you a clue

**Comment by karen20ced4** [comment]
a see a lot of errors in avideo.log, i put here a clean log , only from loading homepage and video page:


[20-Jul-2024 22:02:49 Europe/Quebec] AVideoLog::ERROR:      MySQL ERROR: [{"file":"\/home\/scratchvideos\/public_html\/objects\/mysql_dal.php","line":328,"function":"log_error"},{"file":"\/home\/scratchvideos\/public_html\/plugin\/Live\/Objects\/LiveTransmitionHistory.php","line":491,"function":"readSql","class":"sqlDAL","type":"::"},{"file":"\/home\/scratchvideos\/public_html\/plugin\/Live\/Live.php","line":2774,"function":"getLatest","class":"LiveTransmitionHistory","type":"::"},{"file":"\/home\/scratchvideos\/public_html\/plugin\/Live\/latestOrLive.php","line":68,"function":"getLatest","class":"Live","type":"::"}] SCRIPT_NAME: /plugin/Live/latestOrLive.php

[20-Jul-2024 22:02:49 Europe/Quebec] AVideoLog::ERROR:      [sqlDAL::readSql] (stmt) is empty SELECT 
            lth.*,
            lt.id as live_transmitions_id,
            lt.categories_id
        FROM live_transmitions_history lth
        LEFT JOIN live_transmitions lt ON lth.users_id = lt.users_id
        WHERE 1=1  AND finished IS NULL  ORDER BY (lth.`key` = '') DESC, lth.created DESC LIMIT 1 with formats  SCRIPT_NAME: /plugin/Live/latestOrLive.php

[20-Jul-2024 22:02:49 Europe/Quebec] AVideoLog::DEBUG:      isLiveAndIsReadyFromKey::key:  checking live_servers_id= Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0 SCRIPT_NAME: /plugin/Live/latestOrLive.php

[20-Jul-2024 22:02:49 Europe/Quebec] AVideoLog::DEBUG:      Live::isKeyLiveInStats key is empty SCRIPT_NAME: /plugin/Live/latestOrLive.php

[20-Jul-2024 22:03:01 Europe/Quebec] AVideoLog::DEBUG:      idToHash error: error:03000082:digital envelope routines::invalid key length

{"id":7,"cipher_algo":"rc4","base":32,"idConverted":"7","hash":"","iv":""} SCRIPT_NAME: /view/modeYoutube.php

[20-Jul-2024 22:03:01 Europe/Quebec] AVideoLog::DEBUG:      idToHash error: error:03000082:digital envelope routines::invalid key length

{"id":7,"cipher_algo":"rc4","base":32,"idConverted":"7","hash":"","iv":""} SCRIPT_NAME: /view/modeYoutube.php

[20-Jul-2024 22:03:01 Europe/Quebec] AVideoLog::DEBUG:      idToHash error: error:03000082:digital envelope routines::invalid key length

{"id":7,"cipher_algo":"rc4","base":32,"idConverted":"7","hash":"","iv":""} SCRIPT_NAME: /view/modeYoutube.php

[20-Jul-2024 22:03:01 Europe/Quebec] AVideoLog::DEBUG:      getCaptchaImage: admin - session_name e3734a49c863b81ee95c74d16069cb53 session_id: boh9pnksqg9o5gjr5vf3bkb093 SCRIPT_NAME: /objects/getCaptcha.php

[20-Jul-2024 22:03:02 Europe/Quebec] AVideoLog::DEBUG:      idToHash error: error:03000082:digital envelope routines::invalid key length

{"id":38,"cipher_algo":"rc4","base":32,"idConverted":"16","hash":"","iv":""} SCRIPT_NAME: /view/videosList.php

[20-Jul-2024 22:03:02 Europe/Quebec] AVideoLog::DEBUG:      idToHash error: error:03000082:digital envelope routines::invalid key length
{"id":36,"cipher_algo":"rc4","base":32,"idConverted":"14","hash":"","iv":""} SCRIPT_NAME: /view/videosList.php

**Comment by karen20ced4** [comment]
yes i check the html code where the ad should be and i see this message <!-- getAdsSideRectangle is empty videos_id=7 :
```
<div class="col-lg-12 col-sm-12 col-xs-12 text-center">
<!-- getAdsSideRectangle is empty videos_id=7, not from live videos_id=7, type=sideRectangle, label=sideRectangle [1] [sideRectangle]` --></div>
    <!-- Line 56 -->
    <div class="row">
        <div class="col-lg-12 col-sm-12 col-xs-12 autoplay text-muted" style="margin: 10px 0;">
            <strong>Up Next</strong>
            <span class="pull-right">
                <span>Autoplay</span>
                <span>
```

**Comment by karen20ced4** [comment]
i put back the old file,  /plugins/ADs/ADs.php and now is show the ads normally. at least I managed to isolate the problem  :). 
I also try to use differet PHP version, 7.4 / 8.1 / 8.2 / 8.3 , no succes

**Comment by DanielnetoDotCom** [comment]
Are you using git pull to update?

**Comment by karen20ced4** [comment]
yes, git pull only for updates

**Comment by DanielnetoDotCom** [comment]
How come you put back the old file, /plugins/ADs/ADs.php?

it may break the git pull

Anyway, please check the option ShowOnVideoPlayerPage in your ads plugin

![image](https://github.com/user-attachments/assets/2d184929-b2e1-4ab6-9e40-e88f4ba6573d)

**Comment by karen20ced4** [comment]
Hello, yes that option is activated, that was my first in checking list :). until now i have no other fix that to put back the old file  /plugins/ADs/ADs.php .  
i move my sites to another server, and is the same, no ads for that location. 
i will try with a fresh install and see if is good. you have a solution to migrate the all the videos into a fresh install ?

thanks for your efforts and help
